src/lm.jl

```diff
@@ -64,7 +64,7 @@ end
 cholfact(x::LinearModel) = cholfact(x.pp)
 
 function fit{LmRespT<:LmResp,LinPredT<:LinPred}(::Type{LinearModel{LmRespT,LinPredT}},
-    X::AbstractMatrix, y::Vector)
+    X::@compat(Union{Matrix,SparseMatrixCSC{FP}}), y::FPVector)
```

Comment:
`SparseMatrixCSC{FP}` will only accept matrices with the abstract type `FP`, and not of any concrete subtypes. Use `T<:FP` as for GLMs.

BTW, it would be great to test sparse matrices (which would have caught this problem).